packages/zowe-explorer/src/job/ZosJobsProvider.ts

```diff
@@ -218,53 +217,45 @@ export class ZosJobsProvider extends ZoweTreeProvider implements IZoweTree<IZowe
      * @param {string} [sessionName] - optional; loads default profile if not passed
      * @param {string} [profileType] - optional; loads profiles of a certain type if passed
      */
-    public async addSession(sessionName?: string, profileType?: string): Promise<void> {
+    public async addSession(sessionName?: string, profileType?: string, provider?: IZoweTree<IZoweTreeNode>): Promise<void> {
         ZoweLogger.trace("ZosJobsProvider.addSession called.");
-        const setting: boolean = SettingsConfig.getDirectValue(globals.SETTINGS_AUTOMATIC_PROFILE_VALIDATION);
-        // Loads profile associated with passed sessionName, default if none passed
-        if (sessionName) {
-            const theProfile: imperative.IProfileLoaded = Profiles.getInstance().loadNamedProfile(sessionName);
-            if (theProfile) {
-                await this.addSingleSession(theProfile);
-            }
-            for (const node of this.mSessionNodes) {
-                const name = node.getProfileName();
-                if (name === theProfile.name) {
-                    await resetValidationSettings(node, setting);
-                }
+        await super.addSession(sessionName, profileType, provider);
```

Comment:
Now I understand why we did this.
Makes our codebase a lot more straightforward and easier to follow.
However, aren't we putting a bit of a burden on extenders by requiring them to implement such methods?

I could be misunderstanding the refactoring altogether though. so please correct me if this is not the case.

Reply:
No new code needs to be implemented, to break this down in a easier to understand way the behavior is as follows:

Scenario where `profile` is added from jobs tree view
1. User is prompted for either all or current tree
2. User selects `all trees`
3. `ZosJobsProvider.addSession` is called
4. Previous function now calls parent function of `ZoweTreeProvider.addSession` (or `super.addSession`)
5. Since all was selected, we enter the `else` case and add for each provider using the newly added static class for accessing all tree providers
6. Same old code runs for adding a profile to a tree view (this code already existed, its just now being used differently since it was moved to new functions and cleaned up a bit) 
7. `addSingleSession` is called per tree provider (each one has it defined in their class)